client/packages/common/src/ui/layout/tables/components/Header/ColumnPicker.tsx

```diff
@@ -101,13 +73,15 @@ export const ColumnPicker = <T extends RecordWithId>({
           <Typography style={{ fontWeight: 700 }}>
             {t('table.show-columns')}
           </Typography>
-          {Object.values(columns)
+          {columns
             .filter(c => !!c.label)
             .map(column => (
               <FormControlLabel
                 key={String(column.key)}
-                checked={isVisible(column)}
-                control={<Checkbox onClick={() => toggleColumn(column)} />}
+                checked={columnDisplayState[column.key] ?? true}
```

Comment:
Again, the fallback ensures that any new columns appear as enabled initially. It doesn't actually matter that they're not explicitly defined in the display state yet -- once they get toggled for the first time, it'll be inserted.